src/components/control/autocompleter.jsx

```diff
@@ -1,28 +1,45 @@
 import PropTypes from 'prop-types';
-import React, { forwardRef, useEffect, useState } from 'react';
+import React, { forwardRef, useEffect, useImperativeHandle, useRef, useState } from 'react';
 import useFetch from '@bloodyaugust/use-fetch';
 import useMounted from '../../hooks/use-mounted.js';
 import Select from './select.jsx';
 import ListOption from '../list-option/list-option.jsx';
 import { API_ROOT } from '../../mocks/mock-constants.js';
 
 const Autocompleter = forwardRef(function Autocompleter(props, ref) {
-  const { execute } = useFetch();
+  const { execute: executeModels } = useFetch();
+  const { execute: executeValue } = useFetch();
   const mounted = useMounted();
-  const [models, setModels] = useState(props.models);
+  const [models, setModels] = useState([]);
+  const [model, setModel] = useState();
+  const selectRef = useRef();
 
   useEffect(() => {
-    if (!mounted.current) return;
+    fetchModels();
+  }, []);
 
-    setModels(props.models);
-  }, [props.models]);
-
-  useEffect(fetchModels, []);
+  useEffect(() => {
+    if (props.value) {
+      executeValue(apiEndpoint('only', props.value))
+        .then((respObj) => {
+          if (mounted.current) {
+            setModel(respObj.json.results.map(result => respObj.json[result.key][result.id])[0]);
+          }
+        })
```

Comment:
The entire benefit of `useFetch` is that it has the `isMounted` behavior built-in right? As in the second arg of the promise returned from `executeValue` should be the same as `mounted`?

Reply:
It's got a few other nifty things but yeah it does have its own `mounted` var. I can clean this up.

Reply:
The other benefits of `use-fetch` basically come down to moving it into a hook API which feels nicer in react land, abstracting away the need for `useEffect` from your component, automatically transforming the payload to `JSON`, and cancelling the request when the parent component un-mounts.